src/datavalues.jl

```diff
@@ -24,8 +24,7 @@ DataValueRowIterator(::Type{NT}, x::S) where {NT <: NamedTuple, S} = DataValueRo
 DataValueRowIterator(::Type{Schema{names, types}}, x::S) where {names, types, S} = DataValueRowIterator{datavaluetype(NamedTuple{names, types}), S}(x)
 
 function datavaluerows(x)
-    r = Tables.rows(x)
-    #TODO: add support for unknown schema
+    r = Tables.rowtable(x)
```

Comment:
@quinnj Why was that changed from ``Tables.rows``? If I understand this correctly, it now means that if someone uses Query.jl with a ``DataFrame`` source, it will first allocate a new vector of named tuples, and then pass that on to Query.jl. That seems super inefficient, and really we would want the old behavior here that the named tuples are created per ``iterate`` call and streamed, right?

I think going back to the old behavior might also solve https://github.com/queryverse/Query.jl/issues/240. I'm still trying to understand what is happening there, but if I load that file and then do ``df |> getiterator |> collect``, everything hangs forever. I'm not at the bottom of what is happening there, but on the other hand, having the ``collect`` call in there seems generally undesirable, so maybe that would just solve the bug.

Reply:
Well, it would be better to figure out what exactly is causing the hang. The change here is intentional, because, as shown in the previous code comment, unknown schema weren't supported by this function. By materializing as a Vector of NamedTuples, it handles that case for us. Maybe we can figure out a different solution altogether that doesn't allocate a new vector while still handling unknown schema.

Reply:
I think it is really important to go back to the old behavior for the ``DataFrame`` and Query.jl combination. For that scenario there shouldn't be any issue about unknown schemas. At the end of the day, this change here is a major regression for the most common usage scenario for Query.jl: this PR now makes a copy of everything in the DataFrame before any query runs, which really defeats the whole design of Query where multiple query operators get fused and intermediate allocations of results are avoided at all costs when possible.

I understand that we in general want to handle the case of unknown schemas here, but as far as I can tell right now we actually don't have any case in the wild where ``datavaluerows`` could be called with an ``x`` that doesn't have a schema, right? At least right now I can't think of any combination of sinks and sources where such a situation would come up. If that is correct, then I think it is an even stronger argument to undo this change here quickly: I don't think it is the right trade-off in the short term to regress the main usage scenario for a benefit in a scenario that can't actually come up right now.

In terms of a solution: can't we just check ``schema(x)===nothing``  at the beginning of this function here and only then allocate an intermediate array? I don't fully understand

Reply:
@davidanthoff I'd say make a PR to revert the problematic line until we find a better solution.

(Thouh AFAICT column vectors are not copied when a data frame is passed, they are just put in a named tuple. But I guess that can be costly for the compiler if there are many columns.)

Reply:
https://github.com/JuliaData/Tables.jl/pull/67